### DISCUSSION TITLE
Updating Marker iconAnchor after initialization

### DISCUSSION THREAD

**Discussion by newmanw** [Leaflet/Leaflet#9394] [Q&A]
I am in a situation in which I pull an icon from a url and set the width to a fixed width.  In this case I do not know the height of the icon until the icon image is loaded in the DOM.  I am trying to update the _iconAnchor_ via _setOptions()_ after initialization.  However, the iconAnchor is not updated.

I have a custom icon class that will load the icon and callback into the custom Marker class with the icon

```
    const img = new Image();
    img.src = '<some dynamic url>';
    img.style.width = this.iconWidth + 'px';

    img.onload = (element) => {
      this.onIconLoad(element); // callback into the marker class when the icon has been loaded
    }
```

From there I am trying to update the iconAnchor
```

fixedWidthIcon({
        iconUrl: options.iconUrl,
        iconWidth: 60,
        onIconLoad: (element) => {
          if (icon) {
            setOptions(this, {
              iconAnchor: new Point(0, element.srcElement.height) // setOptions called, but anchor/marker is not adjusted
            })
          }
        }
      });
    }
```

**Comment by Falke-Design** [comment]
Only changing the options will not trigger any modifications. 

You can create a completly new icon and set it with `marker.setIcon(newIcon)` or you do following with internal functions:
```
	marker.options.icon.options.iconAnchor = [20,20]
	marker.options.icon._setIconStyles(marker._icon, 'icon')
	marker.options.icon._setIconStyles(marker._shadow, 'shadow') // if existing
```
